### PR TITLE
Generate release asset sha256 checksums.txt

### LINKED ISSUE #3378: Provides binaries checksums as part of release artifacts
### Is your feature request related to a problem? Please describe.

Hello,

k0s doesn't provides binary checksum to ensure the downloaded k0s binary is correct. It would be nice if they could be added in release artifacts. In some of my personal Ansible roles, I usually add a checksum link to a remote file to ensure the downloaded binary is correct, here is an example with `etcd` : https://github.com/f-bn/etcd-role/blob/main/defaults/main.yml#L16C2-L16C2.

I would like to be able to do the same thing with my `k0s` role :)

### Describe the solution you would like

Provides checksums as a file like many projects : 
- https://github.com/etcd-io/etcd/releases/download/v3.5.9/SHA256SUMS
- https://github.com/prometheus/prometheus/releases/download/v2.37.9/sha256sums.txt
- https://github.com/distribution/distribution/releases/download/v2.8.2/registry_2.8.2_linux_amd64.tar.gz.sha256

### Describe alternatives you've considered

None for now

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kke**
## Description

Generate release asset checksums.txt during release workflow.

Fixes #3378 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [ ] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [X] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by kke** [comment]
It's a bit difficult to validate if it works or not without making a release 🤔

**Review by mikhail-sakhnov — APPROVED** [review]

**Comment by mikhail-sakhnov** [comment]
> It's a bit difficult to validate if it works or not without making a release 🤔

you could validate it by temporary changing triggers, for example to run it on pull requests

**Comment by kke** [comment]
That would require changing the release asset upload part to maybe regular asset upload 🤔

**Review by juanluisvaladas — APPROVED** [review]

**Comment by juanluisvaladas** [comment]
We can just push one tag without acutally making the release, this would trigger the release pipeline but the release won't be published.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -342,6 +342,32 @@ jobs:
...
+          out-file-path: release_assets
+      - name: Generate checksums
+        run: |
+          sha256sum -b release_assets/* | sed 's/release_assets\///' > checksums.txt
+      - name: Upload checksums
+        uses: shogo82148/actions-upload-release-asset@v1.6.5
+        with:
+          upload_url: ${{ needs.release.outputs.upload_url }}
+          asset_path: checksums.txt
+          asset_name: checksums.txt
```
Comment (twz123) [inline .github/workflows/release.yml:367]: I think the name `sha256sums.txt` would be better. It'd be clearer which hashing algorithm has been used that way.

Comment (kke) [reply .github/workflows/release.yml]: done

**Review by twz123 — APPROVED** [review]
Just realizing that this might produce wrong hash sums when the files get corrupted during the download. Would have been better to hash the files before uploading. But let's keep that as a potential future improvement :slightly_smiling_face:

**Pull request merged**